src/cockpit/samples.py

```diff
@@ -301,6 +302,55 @@ def sample(self, samples: Samples) -> None:
                     samples['cgroup.cpu.usage'][cgroup] = usage_nsec / 1000000
 
 
+class CGroupDiskIO(Sampler):
+    descriptions = [
+        SampleDescription('disk.cgroup.read', 'bytes', 'counter', True),
+        SampleDescription('disk.cgroup.written', 'bytes', 'counter', True),
+    ]
+
+    @staticmethod
+    def get_cgroup_name(fd: int) -> str:
+        with Handle.open('cgroup', os.O_RDONLY, dir_fd=fd) as cgroup_fd:
+            cgroup_name = os.read(cgroup_fd, 2048).decode().strip()
+
+            # Skip leadin ::0/
+            return cgroup_name[4:]
+
+    @staticmethod
+    def get_proc_io(fd: int) -> Tuple[int, int]:
+        with Handle.open('io', os.O_RDONLY, dir_fd=fd) as io_fd:
+            data = os.read(io_fd, 4096)
+
+            match = re.search(rb'\bread_bytes: (?P<read>\d+)\nwrite_bytes: (?P<write>\d+)', data)
```

Comment:
Is it guaranteed that the read bytes will come before the write bytes?

Can you please pre-compile the RE in static scope?  We do this once a second for every single process on the system.